backend/test/test_common/unit/utils/test_type_conversion_utils.py

```diff
@@ -213,5 +217,272 @@ def test__convert_pandas_series_to_numpy__int64_to_int32_with_nans_throws_error(
+    {
+        "data": data,
+        "expected_encoding_dtype": np.uint8,
+        "expected_schema_hint": {"type": "boolean"},
+    }
+    for data in [
+        np.array([0, 1, 0, 1], dtype=np.bool_),
+        pd.Series(np.array([0, 1, 0, 1], dtype=np.bool_)),
+        # pd.Index with bools doesn't really make any sense...and becomes dtype=object
+    ]
+]
+
+int_OK_cases = [
+    {
+        "data": data,
+        "expected_encoding_dtype": np.int32,
+        "expected_schema_hint": {"type": "int32"},
+    }
+    for dtype in [np.int8, np.uint8, np.int16, np.uint16, np.int32, np.uint32, np.int64, np.uint64]
+    for data in [
+        np.arange(0, 1000, dtype=dtype),
+        pd.Series(np.arange(0, 1000, dtype=dtype)),
+        pd.Index(np.arange(0, 1000, dtype=dtype)),
+        sparse.csr_matrix((10, 100), dtype=dtype),
+    ]
+]
+
+float_OK_cases = [
+    {
+        "data": data,
+        "expected_encoding_dtype": np.float32,
+        "expected_schema_hint": {"type": "float32"},
+        "logs": None if data.dtype != np.float64 else {"level": logging.WARNING, "output": "may lose precision"},
+    }
+    for dtype in [np.float16, np.float32, np.float64]
+    for data in [
+        np.arange(-128, 1000, dtype=dtype),
+        pd.Series(np.arange(-128, 1000, dtype=dtype)),
+        pd.Index(np.arange(-129, 1000, dtype=dtype)),
+        np.array([-np.nan, np.NINF, -1, np.NZERO, 0, np.PZERO, 1, np.PINF, np.nan], dtype=dtype),
+        np.array([np.finfo(dtype).min, 0, np.finfo(dtype).max], dtype=dtype),
+        sparse.csr_matrix((10, 100), dtype=dtype),
+    ]
+]
+
+
+numeric_ERR_cases = [
+    {
+        "data": data,
+        "throws": TypeError,
+    }
+    for data in [
+        np.array([np.iinfo(np.int64).min, np.iinfo(np.int64).max], dtype=np.int64),
+        np.array([np.iinfo(np.uint64).min, np.iinfo(np.uint64).max], dtype=np.uint64),
+        np.array([np.iinfo(np.uint32).min, np.iinfo(np.uint32).max], dtype=np.uint32),
+    ]
+]
+
+
+string_OK_cases = [
+    {
+        "data": data,
+        "expected_encoding_dtype": np.dtype(str),
+        "expected_schema_hint": {"type": "string"},
+    }
+    for data in [
+        np.array(["a", "b", "c"]),
+        np.array(["a", "b", "c"], dtype="object"),
+        pd.Series(["a", "b", "c"]),
+        pd.Index(["a", "b", "c"]),
+        np.array(["a", [], {}, None, True, False, 383.2], dtype="object"),
+    ]
+]
+
+category_nonnumeric_OK_cases = [
+    {
+        "data": data,
+        "expected_encoding_dtype": np.dtype(str),
+        "expected_schema_hint": {"type": "categorical", "categories": data.dtype.categories.to_list()},
+    }
+    for data in [
+        pd.Series(["a", "b", "c"], dtype="category"),
+        pd.Series(["a", "b", "c", 0, 1, 2], dtype="category"),
+        pd.Series(["a", "b", "c"], dtype="category").cat.remove_categories(["b"]),
+        pd.Series(["a", "b", "c", 0, 1, 2], dtype="category").cat.remove_categories(["b", 0]),
+    ]
+]
+
+category_numeric_OK_cases = [
```

Comment:
There was a comment in the original `can_cast_to_int32` function that said roughly:

"If the array is categorical, then we need to order the array values so that functions min and max that occur later, can function. They do not function on unordered categories."

Can we add a test case to verify that the code still works even though those lines were removed?

Reply:
I believe this was a comment added by your earlier PR, and I was unable to find any instance where the stated constraint was true.  Given that reordering changes the underlying type, it seemed unwise to do so without a clear need and API contract.  Looking at the bug report that triggered the original PR, I could not see a reason this change was required.

Do you recall why you added this?  I definitely recommend against it unless there is a corner case I am unaware of.

Reply:
Huh. I can't remember why I added it unfortunately.